support/on-prem-bootstrap.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+
+# The purpose of this script is to download the latest stable version of every
+# package in the core-plans repo, tar them up, and upload to S3. It also supports
+# downloading the archive from S3, extracting it, and uploading to a new depot.
+
+set -e
+
+help() {
+    echo "Usage: on-prem-bootstrap.sh {create-archive|populate-depot <DEPOT_URL>}"
+}
+
+exists() {
+  if command -v $1 >/dev/null 2>&1
+  then
+    return 0
+  else
+    return 1
+  fi
+}
+
+required_tools=( aws git curl jq )
+required_vars=( AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY HAB_AUTH_TOKEN )
+upstream_depot="https://bldr.habitat.sh"
+bucket="habitat-on-prem-builder-bootstrap"
+bootstrap_file="on-prem-bootstrap.tar.gz"
+
+for tool in "${required_tools[@]}"
```

Comment:
We should only need to depend on curl for populate_depot, so this could be two checks...